lib/learn_to_rank/ranker_status.rb

```diff
@@ -0,0 +1,103 @@
+require "aws-sdk-sagemaker"
+require "httparty"
+
+module LearnToRank
+  class RankerStatus
+    attr_reader :errors
+
+    def initialize(timeout: 0.1)
+      @errors = [] # Strings
+      @timeout = timeout
+      @healthy = check_health
+    end
+
+    def healthy?
+      @healthy && errors.none?
+    end
+
+  private
+
+    attr_reader :timeout
+
+    GOOD_STATES = %w(AVAILABLE InService Updating Creating SystemUpdating RollingBack).freeze
+    GOOD_STATUSES = %w(OK).freeze
+
+    def check_health
+      begin
+        reranker_healthy
+      rescue StandardError => e
+        @errors << "#{e.class}: #{e.message}"
+        false
+      end
+    end
+
+    class RankerServerError < StandardError
+      attr_reader :message
+      def initialize(message: nil)
+        @message = message # String
+      end
+    end
+    class StatusRequestFailed < RankerServerError; end
+    class StatusResponseInvalid < RankerServerError; end
+    class ModelUndefined < RankerServerError; end
+    class ModelStateUnhealthy < RankerServerError; end
+    class ModelStatusUnhealthy < RankerServerError; end
+
+    def reranker_healthy
+      endpoint = ENV["TENSORFLOW_SAGEMAKER_ENDPOINT"]
+      endpoint ? sagemaker_healthy? : container_healthy?
+    end
+
+    def sagemaker_healthy?
+      # TODO:
+      true
+    end
+
+    def container_healthy?
+      url = "http://#{tensorflow_serving_ip}:8501/v1/models/ltr"
+      options = {
+        headers: { "Content-Type" => "application/json" },
+        timeout: timeout, # seconds
+      }
+      response = HTTParty.get(url, options)
+      validate_response!(response)
+      model = JSON.parse(response.body).fetch("model_version_status", []).last
+      validate_model_healthy!(model)
+      true
+    rescue HTTParty::Error, SocketError, Timeout::Error => e
+      raise RankerServerError.new(message: e.message)
+    end
+
+    def tensorflow_serving_ip
```

Comment:
Can we re-use the method from `LearnToRank::Ranker` here?

Reply:
Yep, I've moved this into a helper.